internal/cli/down.go

```diff
@@ -100,14 +106,59 @@ func (c *downCmd) down(ctx context.Context, downDeps DownDeps, args []string) er
 	return nil
 }
 
+func deleteK8sManifests(ctx context.Context, manifests []model.Manifest, updateSettings model.UpdateSettings, downDeps DownDeps, deleteNamespaces bool) error {
+	nodes := []*dependencyNode{}
+
+	for i := range manifests {
+		manifest := manifests[len(manifests)-i-1]
+
+		nodes = append(nodes, &dependencyNode{
+			manifest:   manifest,
+			dependents: []*dependencyNode{},
+		})
+	}
+
+	for _, node := range nodes {
+		for _, resourceDep := range node.manifest.ResourceDependencies {
+			for _, other := range nodes {
```

Comment:
would it make sense to have a map of nodes by name? simply to make the lookup here easier to read / more ergonomic